x-pack/legacy/plugins/snapshot_restore/public/app/sections/home/recovery_list/recovery_table/shards_table.tsx

```diff
@@ -0,0 +1,252 @@
+      }),
+    },
+    {
+      field: 'bytesTotal',
+      name: i18n.translate('xpack.snapshotRestore.recoveryList.shardTable.bytesColumnTitle', {
+        defaultMessage: 'Bytes',
+      }),
+      render: (
+        bytesTotal: SnapshotRecoveryShard['bytesTotal'],
+        { bytesRecovered, bytesPercent }: SnapshotRecoveryShard
+      ) => {
+        return (
+          <EuiToolTip
+            position="top"
+            content={i18n.translate(
+              'xpack.snapshotRestore.recoveryList.shardTable.bytesTooltipLabel',
+              {
+                defaultMessage: 'Recovered: {recovered}, Total: {total}',
+                values: {
+                  recovered: bytesRecovered,
+                  total: bytesTotal,
+                },
+              }
+            )}
+          >
+            <EuiText size="xs" textAlign="center" style={{ width: '100%' }}>
+              <EuiProgress
+                value={bytesTotal === 0 ? 1 : bytesRecovered}
+                max={bytesTotal === 0 ? 1 : bytesTotal}
+                size="xs"
+              />
+              <EuiSpacer size="xs" />
+              {bytesPercent}
+            </EuiText>
+          </EuiToolTip>
+        );
+      },
+    },
+    {
+      field: 'filesTotal',
+      name: i18n.translate('xpack.snapshotRestore.recoveryList.shardTable.filesColumnTitle', {
+        defaultMessage: 'Files',
+      }),
+      render: (
+        filesTotal: SnapshotRecoveryShard['filesTotal'],
+        { filesRecovered, filesPercent }: SnapshotRecoveryShard
+      ) => {
+        return (
+          <EuiToolTip
+            position="top"
+            content={i18n.translate(
+              'xpack.snapshotRestore.recoveryList.shardTable.filesTooltipLabel',
+              {
+                defaultMessage: 'Recovered: {recovered}, Total: {total}',
+                values: {
+                  recovered: filesRecovered,
+                  total: filesTotal,
+                },
+              }
+            )}
+          >
+            <EuiText size="xs" textAlign="center" style={{ width: '100%' }}>
+              <EuiProgress
+                value={filesTotal === 0 ? 1 : filesRecovered}
+                max={filesTotal === 0 ? 1 : filesTotal}
+                size="xs"
+              />
+              <EuiSpacer size="xs" />
+              {filesPercent}
+            </EuiText>
+          </EuiToolTip>
+        );
+      },
+    },
+    {
+      field: 'translogTotal',
+      name: i18n.translate('xpack.snapshotRestore.recoveryList.shardTable.translogColumnTitle', {
+        defaultMessage: 'Translog',
+      }),
+      render: (
+        translogTotal: SnapshotRecoveryShard['translogTotal'],
+        { translogRecovered, translogPercent }: SnapshotRecoveryShard
+      ) => {
+        return (
+          <EuiToolTip
+            position="top"
+            content={i18n.translate(
+              'xpack.snapshotRestore.recoveryList.shardTable.filesTooltipLabel',
```

Comment:
Just a Nit; Maybe this can be abstracted into a separate function since its being used multiple times.

Reply:
Hey I had the same comment! See my review below.